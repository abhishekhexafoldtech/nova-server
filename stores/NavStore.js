import { defineStore } from "pinia";
import bootstrap from "bootstrap/dist/js/bootstrap.min.js";

export const useNavStore = defineStore("NavStore", {
  // state
  state: () => {
    return {
      bootstrap: bootstrap,
      isSidenavCollapsed: false,
      showConfig: false,
      sidenavType: 'bg-white',
      navFixed: false,
      darkMode: false,
      miniNotification: false,
      breadCrumb: "Nova-Gas"
    };
  },

  actions: {
    toggleSidenavCollapsing() {
      this.isSidenavCollapsed = !this.isSidenavCollapsed;
    },
    toggleShowConfig() {
      this.showConfig = !this.showConfig;
    },
    toggleSidebarColor(payload) {
      this.sidebarColor = payload;
    },
    setSidenavType(payload) {
      this.sidenavType = payload;
    },
    toggleDarkMode() {
      this.darkMode = !this.darkMode;
    },
    toggleNavFixed() {
      this.navFixed = !this.navFixed;
    },
    setBreadCrumbs(data) {
      this.breadCrumb = data
    }
  },
});

