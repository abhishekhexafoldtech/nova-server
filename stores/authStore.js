import { defineStore } from 'pinia';
import { getToken, removeToken, setToken } from '@/utils/auth'
import {  getUserInfo, loginByWeb, logoutByWeb } from '@/api/login'

export const useAuthStore = defineStore("auth", {

  state: () => ({
    token: getToken(),
    user_info: {}, // complete user details
    page_refreshing: true,
  }),

  getters: {
    getLoginToken() {
      return this.token
    },
  },

  actions: {
	// Login
	Login(userCredInfo) {
		return new Promise((resolve, reject) => {
			// make login data set
			let data = {
				email: userCredInfo.email,
				phone: userCredInfo.phone,
				password: userCredInfo.password,
			}
			// Lets fire login with info
			loginByWeb(data).then(response => {
				if(response.status == 200) {
					let responseData = response.data
					// console.log("responseData", responseData)
					let go = true
						if (go) {
							// SITE TOKEN
							this.token = responseData.access_token
							this.user_info = responseData.data
							setToken(responseData.access_token)

							const person = { 
									"id": responseData.data.id,
									"first_name": responseData.data.first_name,
									"last_name": responseData.data.last_name,
									"email": responseData.data.email,
									"phone": responseData.data.phone,
								}
							
							// console.log('person', person)
							localStorage.setItem("user", JSON.stringify(person));

							resolve()
						}
					}
						else {
						alert('Something went wrong!')
						reject()
					}
			}).catch(error => {
				reject(error)
			})
		})
	},

	// Logout
	LogOut() {
		return new Promise((resolve, reject) => {
			this.token = ""
			removeToken()
			localStorage.clear();
			resolve()
			// logoutByWeb().then((response) => {
			// 	this.token = ""
			// 	removeToken()
			// 	resolve(response)
			// }).catch(error => {
			// 	console.log(error)
			// 	reject(error)
			// })
			// .finally(() => {
			// })
		})
	},

	// Navbar loaded or refreshed : both cases fired
	pageLoadedORRefreshed() {
		// Get user info
		return new Promise((resolve) => {
			this.page_refreshing = true;
			// Get user info
			// getUserInfo().then((response) => {
			// this.user_info = response.data.data
			// 	resolve(response);
			// })
			// Get user info from local storage
			let temp_data = localStorage.getItem("user")
			this.user_info = JSON.parse(temp_data)
		});
	},
  },
});
