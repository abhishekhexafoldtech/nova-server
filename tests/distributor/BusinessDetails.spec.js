import { shallowMount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import { pinia } from "@/tests/mockpinia";
import { useNavStore } from "@/stores/NavStore";
import BusinessDetails from "@/pages/distributor/add-distributor/business-details.vue";

describe("BusinessDetails Component", () => {
  const navStore = useNavStore(pinia);
  it("renders all input fields", () => {
    const wrapper = shallowMount(BusinessDetails);

    //  BusinessDetails component uses Element Plus UI elements, you can use specific queries to check for the presence of those elements.

    //  Check if an Element Plus input element with a label name is present
    expect(wrapper.find('[label="Name of Business"]').exists()).toBe(true);

    expect(wrapper.find('[label="Company Registration Number"]').exists()).toBe(true);

    expect(wrapper.find('[label="House No. / Street"]').exists()).toBe(true);

    expect(wrapper.find('[label="GPS Code"]').exists()).toBe(true);

    expect(wrapper.find('[label="Area"]').exists()).toBe(true);

    expect(wrapper.find('[label="Region"]').exists()).toBe(true);

    expect(wrapper.find('[label="District"]').exists()).toBe(true);

    expect(wrapper.find('[label="Post Code"]').exists()).toBe(true);

    expect(wrapper.find('[label="GPS Address"]').exists()).toBe(true);

    // Assert that the rendered HTML matches a file snapshot if you're using snapshot testing.
    expect(wrapper.html()).toMatchSnapshot();
  });

  //check save button present or not 
  it("renders the Save button",async () => {
    const wrapper = shallowMount(BusinessDetails);

    // Check if the "Continue" button is present
    expect(wrapper.find(".btn-primary").text()).toBe("Save");
   await wrapper.find(".btn-primary").trigger('click')
  });

  it("renders the SingleFileUpload Component", () => {
    const wrapper = shallowMount(BusinessDetails);
     // Find the <SingleFileUpload> component within the parent component
    const singleFileUpload = wrapper.findComponent({ name: 'SingleFileUpload' });

    // Assert that the SingleFileUpload component exists within the parent component
    expect(singleFileUpload.exists()).toBe(true);
  });

  it('renders the Certificate Of registration component', () => {
    const wrapper = shallowMount(BusinessDetails); 

    // Find the SingleFileUpload component based on its id
    const singleFileUpload = wrapper.find('#CertificateOfregistration');

    // Use assertions to verify that the SingleFileUpload component exists
    expect(singleFileUpload.exists()).toBe(true);
  });

  it('renders the Certificate Of Commencement', () => {
    const wrapper = shallowMount(BusinessDetails); 

    // Find the SingleFileUpload component based on its id
    const singleFileUpload = wrapper.find('#CertificateOfCommencement');

    // Use assertions to verify that the SingleFileUpload component exists
    expect(singleFileUpload.exists()).toBe(true);
  });

});



