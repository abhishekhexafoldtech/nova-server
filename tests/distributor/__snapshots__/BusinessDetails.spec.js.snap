// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BusinessDetails Component > renders all input fields 1`] = `
"<div data-v-e541af4d=\\"\\">
  <section data-v-e541af4d=\\"\\" class=\\"edit_mang_wrap\\">
    <div data-v-e541af4d=\\"\\" class=\\"mang_inner comp_inner\\">
      <div data-v-e541af4d=\\"\\" class=\\"mang_title\\">
        <h3 data-v-e541af4d=\\"\\" class=\\"mb-3\\">Business details</h3>
        <p data-v-e541af4d=\\"\\">Enter your business details</p>
      </div>
      <el-row data-v-e541af4d=\\"\\" gutter=\\"50\\">
        <el-col data-v-e541af4d=\\"\\" xs=\\"24\\" sm=\\"24\\" md=\\"12\\" lg=\\"11\\" xl=\\"11\\">
          <el-form data-v-e541af4d=\\"\\" label-position=\\"top\\" label-width=\\"100px\\" model=\\"[object Object]\\" rules=\\"[object Object]\\">
            <div data-v-e541af4d=\\"\\" class=\\"comp_form_box\\">
              <!-- Name of Business -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"Name of Business\\" prop=\\"name_of_bussiness\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"Name of business\\"></el-input>
                </el-form-item>
              </div><!-- Company Registration Number -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"Company Registration Number\\" prop=\\"company_registration_number\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"Company registration number\\"></el-input>
                </el-form-item>
              </div><!-- House No. / Street -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"House No. / Street\\" prop=\\"house_number\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"House no., Street\\"></el-input>
                </el-form-item>
              </div><!-- GPS Code -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"GPS Code\\" prop=\\"gps_code\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"GPS code\\"></el-input>
                </el-form-item>
              </div><!-- Area -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"Area\\" prop=\\"area\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"Area\\"></el-input>
                </el-form-item>
              </div><!-- Region -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"Region\\" prop=\\"region\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"Region\\"></el-input>
                </el-form-item>
              </div><!-- District -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"District\\" prop=\\"district\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"District\\"></el-input>
                </el-form-item>
              </div><!-- Post Code -->
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"Post Code\\" prop=\\"post_code\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"Post code\\"></el-input>
                </el-form-item>
              </div>
            </div><!-- GPS address -->
            <div data-v-e541af4d=\\"\\" class=\\"comp_form_box\\">
              <h5 data-v-e541af4d=\\"\\" class=\\"cf_title\\">Enter your locationâ€™s GPS address</h5>
              <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
                <el-form-item data-v-e541af4d=\\"\\" label=\\"GPS Address\\" prop=\\"gps_address\\">
                  <el-input data-v-e541af4d=\\"\\" modelvalue=\\"\\" placeholder=\\"GPS address\\"></el-input>
                </el-form-item>
              </div><img data-v-e541af4d=\\"\\" class=\\"gh_post\\" src=\\"/assets/svg/ghanapost.svg\\" alt=\\"GhanaPost\\">
            </div>
          </el-form>
        </el-col>
        <el-col data-v-e541af4d=\\"\\" xs=\\"24\\" sm=\\"24\\" md=\\"12\\" lg=\\"8\\" xl=\\"8\\">
          <div data-v-e541af4d=\\"\\" class=\\"comp_form_box add_dist_up w355\\">
            <!-- Certificate of Registration -->
            <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
              <p data-v-e541af4d=\\"\\">Certificate of Registration</p>
              <single-file-upload-stub data-v-e541af4d=\\"\\" iconclass=\\"iconClass\\" progress=\\"true\\" id=\\"CertificateOfregistration\\" value=\\"[object Object]\\"></single-file-upload-stub>
            </div>
            <!--v-if-->
          </div><!-- Certificate of Commencement -->
          <div data-v-e541af4d=\\"\\" class=\\"comp_form_box add_dist_up w355\\">
            <div data-v-e541af4d=\\"\\" class=\\"fieldrow\\">
              <p data-v-e541af4d=\\"\\">Certificate of Commencement</p>
              <single-file-upload-stub data-v-e541af4d=\\"\\" iconclass=\\"iconClass\\" progress=\\"true\\" id=\\"CertificateOfCommencement\\" value=\\"[object Object]\\"></single-file-upload-stub>
            </div>
            <!--v-if-->
          </div>
        </el-col>
      </el-row><!-- cancel and save button -->
      <div data-v-e541af4d=\\"\\" class=\\"comp_footer\\">
        <nuxt-link data-v-e541af4d=\\"\\" class=\\"btn btn-default\\" to=\\"/distributor\\">Cancel</nuxt-link><button data-v-e541af4d=\\"\\" class=\\"btn btn-primary\\">Save</button>
      </div>
    </div>
  </section>
  <div data-v-e541af4d=\\"\\" class=\\"row\\">
    <!-- Dialog box code -->
    <success-dialog-stub data-v-e541af4d=\\"\\" leftbuttonname=\\"Cancel\\" rightbuttonname=\\"Send email\\" dialogtitle=\\"Admin added successfully\\" dialogimage=\\"/assets/svg/admin_mail.svg\\" dialogtext=\\"Send temporary logic credentials to admin\\" dialogvisible=\\"false\\"></success-dialog-stub>
  </div>
</div>"
`;
